enumerables.rb

```diff
@@ -0,0 +1,225 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    while index < size
+      if is_a? Array
+        yield self[index]
+      elsif is_a? Range
+        yield to_a[index]
+      end
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    (0...size).each do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    values = []
+    my_each do |val|
+      values.push(val) if yield(val)
+    end
+    values
+  end
+
+  def my_all?(value = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    if block_given?
+      values = true
+      my_each { |val| values = false unless yield(val) }
+      values
+    elsif !block_given? && value.is_a?(Numeric)
+      values = true
+      my_each { |val| values = false if val != value }
+      values
+    elsif !block_given? && value.is_a?(String)
+      values = true
+      my_each { |val| values = false if val != value }
+      values
+    elsif !block_given? && value.is_a?(Regexp)
+      values = true
+      my_each { |val| values = false if val }
+      values
+    elsif !block_given? && value.is_a?(Class)
+      values = true
+      my_each { |val| values = false unless val.is_a?(value) }
+      values
+    elsif include?(nil) || include?(false)
+      false
+    elsif !block_given? && value.nil?
+      true
+    end
+  end
+
+  def my_any?(value = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    if block_given?
+      values = false
+      my_each { |val| values = true if yield(val) }
+      values
+    elsif !block_given? && value.is_a?(Regexp)
+      false
+    elsif !block_given? && value.is_a?(Numeric)
+      values = false
+      my_each { |val| values = true if val == value }
+      values
+    elsif !block_given? && value.is_a?(String)
+      values = false
+      my_each { |val| values = true if val == value }
+      values
+    elsif !block_given? && value.is_a?(Class)
+      values = false
+      my_each { |val| values = true if val.is_a?(value) }
+      values
+    elsif !block_given? && value.nil?
+      values = false
+      my_each { |val| values = true if val }
+      values
+    else
+      true
+    end
+  end
+
+  def my_none?(value = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    if block_given?
+      values = true
+      my_each { |val| values = false if yield(val) }
+      values
+    elsif !block_given? && value.is_a?(Regexp)
+      return true
```

Comment:
This part of your method is not working. When you check something like this: ['string', 'ring', 'spring'].my_all?(/ing/ should return false, but it's returning true.